pytorch_lightning/utilities/cli.py

```diff
@@ -849,6 +860,17 @@ def _prepare_subcommand_kwargs(self, subcommand: str) -> Dict[str, Any]:
             fn_kwargs["datamodule"] = self.datamodule
         return fn_kwargs
 
+    def _set_seed(self) -> None:
+        """Sets the seed."""
+        config_seed = self._get(self.config, "seed_everything")
+
+        if isinstance(config_seed, bool) and config_seed:
+            config_seed = _select_seed_randomly()
```

Comment:
why are we setting it randomly though? shouldn't we choose a fixed one, like 0? Otherwise subsequent runs will always be different, unless you pass in a config file.

Reply:
But we want the seed of every run to be different. That is the normal behavior, that you also have when setting no seed (numpy and pytorch then draw random seeds). What we want here is to automatically log them in the config.yaml, so that all experiments that were done with the LightningCLI are automatically reproducible by using the config.yaml.